lib/secrets.rb

```diff
@@ -21,17 +22,22 @@ def initialize(vault_address:, pemfile_path:, ssl_verify:, annotations:, service
     @api_url = api_url
 
     Vault.configure do |config|
-      config.ssl_pem_file = pemfile_path
       config.ssl_verify = ssl_verify
       config.address = vault_address
       config.ssl_timeout  = 3
       config.open_timeout = 3
       config.read_timeout = 2
     end
 
-    # fetch vault token by authenticating with given pem
-    response = http_post(File.join(Vault.address, CERT_AUTH_PATH), ssl_verify: ssl_verify, pem: pemfile_path)
-    Vault.token = JSON.parse(response).fetch("auth").fetch("client_token")
+    # check and see if the authfile is a pem or a token,
+    # then act accordingly
+    begin
```

Comment:
we could, but getting openssl to load on it seems like a pretty good check, no?  It'll raise the OpenSSL::X509::CertificateError if it doesn't like it.